lua/lspconfig/server_configurations/ruby_ls.lua

```diff
@@ -39,6 +39,68 @@ standalone executable.
 group :development do
   gem "ruby-lsp", require: false
 end
+```
+
+Neovim implements the `textDocument/publishDiagnostic` specification
+for the diagnostics feature. Where the server notifies the client with this event.
+
+However, ruby-lsp since the version `0.2.2` does not support this
+specification for diagnostics.
+Instead, it implements the `textDocument/pullDiagnostic` specification,
+that expects the client to send `textDocument/diagnostic` requests to the server
```

Comment:
Instead of adding a huge amount of documentation (for one lang server....), do we need a feature request in Nvim core? Why does it only work with old versions of ruby-lsp ? Is Nvim LSP missing a standard LSP feature?

Reply:
>  Is Nvim LSP missing a standard LSP feature?

this is feature in lsp 3.17 .. as far as i know we missing some lsp feature like `ontypeformat` `workspace/diagnostic` `workspace/diagnostic/refresh` `textDocument/diagnostic` etc ..

Reply:
@justinmk You are right about nvim core and lspconfig missing some features from LSP, but even if the feature get requested, it will have a long roadmap to wait for until `lspconfig` be able to work again as the current documentation is describing.
I forgot to make the feature request as recommended by @glepnir previously, but I think that in the meantime, we would like to complement the documentation for this lang server, just because if someone follow the instructions on it will not get lspconfig working as described.

That is because the current versions for this lang server implements only [Pull Diagnostic](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_pullDiagnostics) from [LSP v3.17](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/) and dropped support to `textDocument/publishDiagnostic`, which is the only currently supported by Nvim.

There are alternatives to these extensive instructions, but any alternative would have a drawback of any sort:
- Make the instructions limit to use a version of `ruby-lsp <= 0.2.1`  
   - That is the highest version the current version of nvim/lspconfig is compatible with
   - That would limit users capacity to benefit of eventual security patches and other features that nvim might be compatible already but are only present on more recent versions for `ruby-lsp`
- Nvim core and lspconfig implement `textDocument/diagnostic` from Lsp v3.17
    - Besides it being the best scenario, some would need to wait for the roadmap for both core and lspconfig to be released to have diagnostic working again as described in the current instructions.
    - This is the only lang server that I'm aware that does not support `textDocument/publishDiagnostic` anymore. I don't know if it's something that will be dropped by LSP and would increase the priority, but so far the priority might be pretty low if ever implemented.